src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -1449,6 +1430,34 @@ private void FinishHandshake(Exception e, AsyncProtocolRequest asyncRequest)
             }
         }
 
+        private void ProcessPendingWrite()
+        {
+            Debug.Assert(_lockWriteState == LockPendingWrite || _lockWriteState == LockHandshake);
```

Comment:
Thanks for adding the assert.  This is good to have, but I'd actually been asking whether we can/should add an assert like:
```C#
Debug.Assert(Monitor.IsEntered(this));
```